nanoFramework.CoreLibrary/System/Reflection/CustomAttributesHelpers.cs

```diff
@@ -3,74 +3,102 @@
                 }
-                else
+                else if (parameterType.IsArray)
                 {
-                    // has default constructor, invoke it
-
-                    // get the types
-                    Type objectType = rawAttributes[i].GetType();
-                    Type paramType = rawAttributes[i + 1].GetType();
-
-                    // get constructor
-                    ConstructorInfo ctor = objectType.GetConstructor(new Type[] { paramType });
+                    // Check for a constructor with a single parameter of the correct array type
+                    constructorInfo = objectType.GetConstructor(new[] { parameterType });
+                    constructorParameters = new[] { attributeDefinitions[i + 1] };
 
-                    // get params
-                    object[] ctorParams = new object[] { rawAttributes[i + 1] };
-
-                    if (ctor is null)
+                    if (constructorInfo is null)
                     {
-                        // give it another try, this time with an array of the parameters types
-                        // rebuild params list 
-                        ctorParams = (object[])rawAttributes[i + 1];
+                        // Check for a constructor with multiple parameters of the correct types
+                        constructorParameters = (object[])attributeDefinitions[i + 1];
 
-                        Type[] paramsTypes = new Type[ctorParams.Length];
+                        var parameterTypes = new Type[constructorParameters.Length];
 
-                        for (int p = 0; p < ctorParams.Length; p++)
+                        for (var p = 0; p < constructorParameters.Length; p++)
                         {
-                            paramsTypes[p] = ctorParams[p].GetType();
+                            parameterTypes[p] = constructorParameters[p].GetType();
                         }
 
-                        ctor = objectType.GetConstructor(paramsTypes);
-
-                        if (ctor is null)
-                        {
-                            throw new InvalidOperationException();
-                        }
+                        constructorInfo = objectType.GetConstructor(parameterTypes);
                     }
+                }
+                else
+                {
+                    // Get a constructor with a single parameter of the correct type
+                    constructorInfo = objectType.GetConstructor(new[] { parameterType });
+                    constructorParameters = new[] { attributeDefinitions[i + 1] };
+                }
 
-                    // invoke constructor with the parameter
-                    attributes[i / 2] = ctor.Invoke(ctorParams);
+                // If we found a constructor invoke it with the parameters
+                if (constructorInfo is not null)
+                {
+                    attributes.Add(constructorInfo.Invoke(constructorParameters));
+                }
+#if DEBUG
```

Comment:
Better to use `Debug.WriteLine()` which accomplishes the same with a simpler approach.